core/auth.ts

```diff
@@ -85,37 +119,72 @@ const config = {
   providers: [
     CredentialsProvider({
       credentials: {
+        type: { type: 'text' },
         email: { label: 'Email', type: 'email' },
         password: { label: 'Password', type: 'password' },
+        jwt: { type: 'text' },
       },
       async authorize(credentials) {
-        const { email, password } = Credentials.parse(credentials);
+        const parsed = Credentials.parse(credentials);
         const cookieStore = await cookies();
         const cartEntityId = cookieStore.get('cartId')?.value;
 
-        const response = await client.fetch({
-          document: LoginMutation,
-          variables: { email, password, cartEntityId },
-          fetchOptions: {
-            cache: 'no-store',
-          },
-        });
+        if (parsed.type === 'password') {
+          const { email, password } = parsed;
+          const response = await client.fetch({
+            document: LoginMutation,
+            variables: { email, password, cartEntityId },
+            fetchOptions: {
+              cache: 'no-store',
+            },
+          });
 
-        if (response.errors && response.errors.length > 0) {
-          return null;
+          if (response.errors && response.errors.length > 0) {
+            return null;
+          }
+
+          const result = response.data.login;
```

Comment:
I think now is the time to extract a lot of this into its own function:
```ts
const login = async (credentials: Partial<Record<"email" | "password", unknown>>) => {
  // ...
}

// ...
async authorize(credentials) {
  const result = await login(credentials);

  if (!result.customer || !result.customerAccessToken) {
    return null;
  }

  return {
    name: `${result.customer.firstName} ${result.customer.lastName}`,
    email: result.customer.email,
    customerAccessToken: result.customerAccessToken.value,
  };
}
```

🍹  I personally would love the `parsed.type` check to be a `switch` statement here.